### PR TITLE
Revamp plugin

### PR DESCRIPTION
**Pull request by tomsquest**
Hi @mqus ,

Here is the rework of the plugin.

Here how it goes: I wanted to add icons of the different editors, then starting to rework parts of the code, then did a lot of changes.

Here is the final work on the plugin.

Mainly:
- icons are displayed
- more editors supported
- code cleaner (matter of taste)

I would like you to review it if you can.
Then I will submit it to the official repo to be shipped with Albert.

Tell me if you have no time, no energy for reviewing.

### CODE REVIEW

**Comment by mqus** [comment]
hey @tomsquest ! I completely missed this PR before opening https://github.com/albertlauncher/python/pull/159.

Thank you for all the time you put in this code. At this point it's pretty clear that I'm standing in the way more than I'm helping, and to be clear, I am very very grateful that you put in the effort and motivation for this!

If you want then you can just host/maintain the plugin yourself (and with your changes I think you can even relicense it, since it is such a complete rewrite). Please let me know if you're willing, I'll then close https://github.com/albertlauncher/python/pull/159 and also point to your code in this repo.

File: __init__.py [hunk 1]
```diff
@@ -57,164 +94,92 @@ def defaultTrigger(self):
...
+            Editor(
+                name="GoLand",
+                icon=plugin_dir + "/goland.svg",
+                config_dir_prefix="JetBrains/GoLand",
+                binaries=["goland"]),
+            Editor(
+                name="IntelliJ IDEA",
+                icon=plugin_dir + "/idea.svg",
+                config_dir_prefix="JetBrains/IntelliJIdea",
+                binaries=["idea"]),
```
Comment (mqus) [inline __init__.py]: the old list was a bit bigger and had more entries here. I compiled them mostly from the archlinux aur which has a huge variety of possible binary names (see where they are linked into the /usr/bin directory here: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intellij-idea-ultimate-edition#n60 , https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intellij-idea-community-edition-no-jre#n34 , https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intellij-idea-ue-eap#n34 , https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intellij-idea-community-edition-git#n68 . maybe even idea.sh (including the `.sh`) would make some sense. I don't even know what other linux distros (eg. Ubuntu ppa's) name their binaries. I think for intellij, at least `intellij-idea-ultimate-edition`, `idea-ce`, `idea-ce-eap` and `idea-ue-eap` should be added (after going through all archlinux packages).

File: __init__.py [hunk 2]
```diff
@@ -57,164 +94,92 @@ def defaultTrigger(self):
...
+            Editor(
+                name="PhpStorm",
+                icon=plugin_dir + "/phpstorm.svg",
+                config_dir_prefix="JetBrains/PhpStorm",
+                binaries=["phpstorm"]),
+            Editor(
+                name="PyCharm",
+                icon=plugin_dir + "/pycharm.svg",
+                config_dir_prefix="JetBrains/PyCharm",
+                binaries=["charm", "pycharm"]),
```
Comment (mqus) [inline __init__.py]: Same as above, found also `pycharm-edu`, `pycharm-ce`, `pycharm-eap`.

File: __init__.py [hunk 3]
```diff
@@ -3,47 +3,84 @@
...
+
+@dataclass
+class Project:
+    name: str
+    path: str
+    last_opened: int
+
+
+@dataclass
+class Editor:
```
Comment (mqus) [inline __init__.py:26]: This is great! Much less hacky than before.

Comment (tomsquest) [reply __init__.py:26]: Thanks!

**Comment by tomsquest** [comment]
Hi @mqus ,

Thanks for the comments!

I will add the missing executables. I did not know Arch or other distrib were packaging Jetbrains editors! I did search where those could come (searching the doc, using the Toolbox...).

When done, I will submit a PR on Albert-plugins repo.
Do you want to be in the list of maintainers?

File: __init__.py (hunk 1 above)
Comment (tomsquest) [reply __init__.py]: Done!

I did not find any "idea.sh" (as often the binary becomes "idea", or "intellij-long-name-edition")

**Comment by tomsquest** [comment]
LGTM!

**Comment by mqus** [comment]
> Hi @mqus ,
> 
> Thanks for the comments!
> 
> I will add the missing executables. I did not know Arch or other distrib were packaging Jetbrains editors! I did search where those could come (searching the doc, using the Toolbox...).

Archlinux AUR and Ubuntu PPAs are non-official packages by users, they can package pretty much everything they want. Since Intellij Community Edition (and maybe also pycharm) are under Apache V2, they could also be packaged officially afaict. I think in arch the AUR install is pretty common but the toolbox has probably more overall users.


> 
> When done, I will submit a PR on Albert-plugins repo. Do you want to be in the list of maintainers?

Hmmm. Actually I'm not sure :D Currently I don't have much motivation to actively contribute anything. But if there is a bug and you're not there then I'll probably look at it. So... yeah, if you want, put me on there :)

**Pull request merged**